pkg/components/state/pluggable.go

```diff
@@ -425,13 +431,23 @@ func toSetRequest(req *state.SetRequest) (*proto.SetRequest, error) {
 		if dataBytes, err = utils.Marshal(reqValue, json.Marshal); err != nil {
 			return nil, err
 		}
+
+		// Add or overwrite metadata with content type
```

Comment:
It would *probably* be ok to modify the request's metadata instance, but I was concerned about changing assumptions about its immutability that other components might have in the request pipeline.  It seemed safer to make a copy.